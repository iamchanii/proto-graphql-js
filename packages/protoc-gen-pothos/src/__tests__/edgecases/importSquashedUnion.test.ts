import { itGeneratesDSLsToMatchSnapshtos } from "../__helpers__/process.test.helper";

describe("import squashed union", () => {
  itGeneratesDSLsToMatchSnapshtos(
    "testapis.edgecases.import_squashed_union.pkg1",
    ["edgecases/import_squashed_union/pkg1/types.pb.pothos.ts"],
  );
  itGeneratesDSLsToMatchSnapshtos(
    "testapis.edgecases.import_squashed_union.pkg2",
    ["edgecases/import_squashed_union/pkg2/types.pb.pothos.ts"],
  );
});
