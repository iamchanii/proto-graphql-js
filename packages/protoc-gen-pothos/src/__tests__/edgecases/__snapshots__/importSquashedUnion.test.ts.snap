// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import squashed union with ts-proto generates pothos DSLs: edgecases/import_squashed_union/pkg1/types.pb.pothos.ts 1`] = `
"// Code generated by protoc-gen-pothos. DO NOT EDIT.
// source: testapis/edgecases/import_squashed_union/pkg1/types.proto

/* eslint-disable */

import { InputObjectRef } from "@pothos/core";
import { builder } from "../../../../builder";
import { OneofMessage1 } from "./testapis/edgecases/import_squashed_union/pkg1/types";

export const OneofMessage1$Ref = builder.objectRef<OneofMessage1>(
  "OneofMessage1",
);
builder.objectType(OneofMessage1$Ref, {
  name: "OneofMessage1",
  fields: (t) => ({
    body: t.expose("body", {
      type: "String",
      nullable: false,
      extensions: { protobufField: { name: "body", typeFullName: "string" } },
    }),
  }),
  isTypeOf: (source) => {
    return (source as OneofMessage1 | { $type: string & {}; }).$type ===
      "testapis.edgecases.import_squashed_union.pkg1.OneofMessage1";
  },
  extensions: {
    protobufMessage: {
      fullName: "testapis.edgecases.import_squashed_union.pkg1.OneofMessage1",
      name: "OneofMessage1",
      package: "testapis.edgecases.import_squashed_union.pkg1",
    },
  },
});

export type OneofMessage1Input$Shape = { body: OneofMessage1["body"]; };

export const OneofMessage1Input$Ref: InputObjectRef<OneofMessage1Input$Shape> =
  builder.inputRef<OneofMessage1Input$Shape>("OneofMessage1Input").implement({
    fields: (t) => ({
      body: t.field({
        type: "String",
        required: true,
        extensions: { protobufField: { name: "body", typeFullName: "string" } },
      }),
    }),
    extensions: {
      protobufMessage: {
        fullName: "testapis.edgecases.import_squashed_union.pkg1.OneofMessage1",
        name: "OneofMessage1",
        package: "testapis.edgecases.import_squashed_union.pkg1",
      },
    },
  });

export type SquashedOneofInput$Shape = {
  msg1?: OneofMessage1Input$Shape | null;
};

export const SquashedOneofInput$Ref: InputObjectRef<SquashedOneofInput$Shape> =
  builder.inputRef<SquashedOneofInput$Shape>("SquashedOneofInput").implement({
    fields: (t) => ({
      msg1: t.field({
        type: OneofMessage1Input$Ref,
        required: false,
        extensions: {
          protobufField: {
            name: "msg1",
            typeFullName:
              "testapis.edgecases.import_squashed_union.pkg1.OneofMessage1",
          },
        },
      }),
    }),
    extensions: {
      protobufMessage: {
        fullName: "testapis.edgecases.import_squashed_union.pkg1.SquashedOneof",
        name: "SquashedOneof",
        package: "testapis.edgecases.import_squashed_union.pkg1",
        options: { "[graphql.object_type]": { squashUnion: true } },
      },
    },
  });

export const SquashedOneof$Ref = builder.unionType("SquashedOneof", {
  types: [OneofMessage1$Ref],
  extensions: {
    protobufOneof: {
      fullName: "testapis.edgecases.import_squashed_union.pkg1.SquashedOneof",
      name: "SquashedOneof",
      package: "testapis.edgecases.import_squashed_union.pkg1",
      fields: [{
        name: "msg1",
        type: "testapis.edgecases.import_squashed_union.pkg1.OneofMessage1",
        options: { "[graphql.object_type]": { squashUnion: true } },
      }],
    },
  },
});
"
`;

exports[`import squashed union with ts-proto generates pothos DSLs: edgecases/import_squashed_union/pkg2/types.pb.pothos.ts 1`] = `
"// Code generated by protoc-gen-pothos. DO NOT EDIT.
// source: testapis/edgecases/import_squashed_union/pkg2/types.proto

/* eslint-disable */

import { InputObjectRef } from "@pothos/core";
import { builder } from "../../../../builder";
import {
  SquashedOneof$Ref,
  SquashedOneofInput$Ref,
  SquashedOneofInput$Shape,
} from "../pkg1/types.pb.pothos";
import { Message } from "./testapis/edgecases/import_squashed_union/pkg2/types";

export const Message$Ref = builder.objectRef<Message>("Message");
builder.objectType(Message$Ref, {
  name: "Message",
  fields: (t) => ({
    msg: t.field({
      type: SquashedOneof$Ref,
      nullable: true,
      resolve: (source) => {
        const value = source.msg?.msg1;
        if (value == null) {
          return null;
        }
        return value;
      },
      extensions: {
        protobufField: {
          name: "msg",
          typeFullName:
            "testapis.edgecases.import_squashed_union.pkg1.SquashedOneof",
        },
      },
    }),
  }),
  isTypeOf: (source) => {
    return (source as Message | { $type: string & {}; }).$type ===
      "testapis.edgecases.import_squashed_union.pkg2.Message";
  },
  extensions: {
    protobufMessage: {
      fullName: "testapis.edgecases.import_squashed_union.pkg2.Message",
      name: "Message",
      package: "testapis.edgecases.import_squashed_union.pkg2",
    },
  },
});

export type MessageInput$Shape = { msg?: SquashedOneofInput$Shape | null; };

export const MessageInput$Ref: InputObjectRef<MessageInput$Shape> = builder
  .inputRef<MessageInput$Shape>("MessageInput").implement({
    fields: (t) => ({
      msg: t.field({
        type: SquashedOneofInput$Ref,
        required: false,
        extensions: {
          protobufField: {
            name: "msg",
            typeFullName:
              "testapis.edgecases.import_squashed_union.pkg1.SquashedOneof",
          },
        },
      }),
    }),
    extensions: {
      protobufMessage: {
        fullName: "testapis.edgecases.import_squashed_union.pkg2.Message",
        name: "Message",
        package: "testapis.edgecases.import_squashed_union.pkg2",
      },
    },
  });
"
`;
